lib/cc/analyzer/container.rb

```diff
@@ -116,7 +153,15 @@ def reap_running_container
       end
 
       def timeout
-        (ENV["CONTAINER_TIMEOUT_SECONDS"] || DEFAULT_TIMEOUT).to_i
+        ENV.fetch("CONTAINER_TIMEOUT_SECONDS", DEFAULT_TIMEOUT).to_i
+      end
+
+      def maximum_output_bytes
+        ENV.fetch("CONTAINER_MAXIMUM_OUTPUT_BYTES", DEFAULT_MAXIMUM_OUTPUT_BYTES).to_i
+      end
+
+      def counter_mutex
+        @counter_mutex ||= Mutex.new
```

Comment:
I can't find a solid reference for this, but `||=` is actually two operations and thus not thread-safe, right? It's hard to imagine what sort of bug this could cause, but it might just be best to initialize this before spawning secondary threads.